### DISCUSSION TITLE
Fish Folk crowdfund campaign launching in one month

### DISCUSSION THREAD

**Discussion by erlend-sh** [fishfolk/jumpy#603] [Announcements]
~~Next month, i.e. end of February or early March~~ End of March, we will be launching a campaign for Fish Folk on the crowdfunding platform [Kickstarter](https://www.kickstarter.com/). 

Leading up to this launch, we will be busy trying to [tune and polish](https://github.com/fishfolk/jumpy/milestone/4) the core game experience (aka minimum viable product) to be demo-ready.

In this document I'll share our WIP pitch page for Kickstarter. I'd love to field questions from people in all of the different communities we fish folk consider ourselves part of: Indie gamedev, Rust, Bevy, open source, and Duck Game(-rs & modders).

You may look at the [actual Kickstarter preview here](https://www.kickstarter.com/projects/erlendsh/fish-folk?ref=4spnjf&token=b46ffe77), but from this point onwards the most up-to-date version of our pitch page will be this GitHub discussion.

---

Fish Folk is a tactical 2D shooter, played by 2-4 players online or on a shared screen. Aim straight left or right; the rest is up to clever movement and positioning in this fish-on-fish brawler!

![PLACEHOLDER](https://ksr-ugc.imgix.net/assets/039/680/253/a40577992b37a82f228fceb70aa72593_original.gif?ixlib=rb-4.0.2&w=680&fit=max&v=1673870915&auto=format&frame=1&q=92&s=1b99d78b2c5efc42a6f23bae01734e05)

PLACEHOLDER

# Key Features

-   Easy to pick up, emphasizing tactics over twitch reaction
    
-   Emergent gameplay (slip on seaweed into a mine)
    
-   Infinitely moddable weapons & cosmetics
    
-   Smart level creation tools with procedural generation
    
-   Playable with bots or friends locally or online
    
-   Tournaments & matchmaking (no locality too small)
    
-   Unique art style by [PixelFrog](https://pixelfrog-assets.itch.io/), shared as CC BY-NC
    
-   Complete OST by [Emerald Jak](https://soundcloud.com/emerald-jak), shared as CC BY-NC
    

# Origin story

We are making a spiritual successor to the cult classic [Duck Game](https://store.steampowered.com/app/312530/Duck_Game/).

Duck Game is outstanding. It deserves the 10/10 rating it can boast on Steam. But like most games, it is no longer in active development. The solo dev [@superjoebob](https://twitter.com/superjoebob) moved on to other ventures.

Way back in 2018 [Erlend](https://github.com/erlend-sh/) and [Jakob](https://www.etterklangstudio.no/) (our two initial co-founders) started racking up countless hours of Duck Game couch-play, especially once we'd built up a catalogue of our own custom 1v1 maps. No game & chill hangout was ever without it from this point onward.

We're still playing it today, but we yearn for a vibrant, living game that is actively evolving together with us as players and creators. [For three years now, since October 2020](https://spicylobster.itch.io/jumpy/devlog/332434/fish-fights-past-present-and-future), we've been steadfastly working on a fitting continuation of the game we love, so that it may go on forever.

# Everlasting Games

A lot of really good computer games have been made in the last few decades. Many of them are designed with such elegant simplicity that, mechanically, they have no expiration date. And yet these wonderfully timeless creations do sadly expire into obsolescence because of the game industry conventions of secrecy, exclusivity and unrelenting newness.

The intention of Spicy Lobster studio (makers of the Fish Folk franchise) is to make [everlasting games](https://spicylobster.itch.io/jumpy/devlog/337996/everlasting-games), holistically designed to last for as long as the game is still being played by a passionate community. We aim to honor this higher bar by following similar development practices as the open software movement with its many long-lived giants such as Linux, WordPress and Blender.

The games we build will last forever because they are:

Open: As long as a game’s inner workings are open, it can be continued, with or without its original creators. – Fish Folk is open source, and all of our development is done entirely in the open, on [GitHub](https://github.com/fishfolk/jumpy) and [Discord](https://discord.gg/4smxjcheE5).

Moddable: By enabling ease of modification and sharing, a game will forever evolve and grow. – We've designed a novel [modding system] based on the open WebAssembly standard.

Cross-platform: Platforms come and go. Existing (equally fully-featured) on multiple platforms leads to a more resilient playerbase. – Our game is built with Rust and Bevy, which we believe to be [the future of game development](https://thefuntastic.com/blog/why-rust-is-the-future-game-dev).

Communal: A connected and tended-to community rooted in friendship is the living essence that keeps any game going. Collective ownership facilitates discourse and feedback between equals. – All collaborators involved with Fish Folk are automatically enrolled in our [Open Gamedev School](https://spicylobster.itch.io/jumpy/devlog/345530/open-gamedev-school), i.e. an opt-in free-for-all community of practitioners.

Sustainable: Continuously developing and servicing a game requires continuous income. Profits should however be in the service of mutual developer-player happiness, and not for outsize individual enrichment. – Spicy Lobster is a [coop studio in the making](https://spicylobster.itch.io/jumpy/devlog/356201/spicy-lobster-open-gamedev-company).

Loving Play: The classics we fell in love with are held in such high regard because they were designed first and foremost in service of play, I.e. they were created out of a love for playfulness. – By relinquishing maximum ownership over our game, with the end-goal of an [exit-to-community](https://www.colorado.edu/lab/medlab/exit-to-community), the people in charge of Fish Folk will be whoever upholds the values and needs of the community at large.

## So.., it's just a remake?

Nah nah nah, it's a lot more than that! In addition to the tried and true core game experience of our esteemed predecessor Duck Game, we're bringing a lot of innovations along for the ride:

### Advanced Modding

After three years of iterating our way through different tech stacks, we've arrived at an [extensible architecture](https://github.com/fishfolk/jumpy/discussions/489) - called the Bones framework - that checks all the boxes for the maximally moddable game we want to make.

### Procedural Levels

Ever made a custom map that's got a great foundation for remixed derivatives, but you just can't be bothered? We've got a working proof-of-concept of a [smart map generator](https://github.com/fishfolk/jumpy/discussions/512) that can take an existing map as an input, and output a collection of remixes, automagically! 💫

### Rollback p2p networking

Fish Folk is first and foremost made for couch-play, but networked play is sometimes the only option we've got. To that end, we've invested a lot of time into making our online experience the best it can be. In collaboration with other savvy people in the Rust ecosystem we've implemented a [netcode system](https://github.com/fishfolk/jumpy/discussions/519) based on the [GGPO](https://en.wikipedia.org/wiki/GGPO) model, popularized in competitive fighting games.

### Character customization & remixing

![](https://ksr-ugc.imgix.net/assets/039/701/040/2749fbd6040a8a9fc6d1757158d8bf28_original.png?ixlib=rb-4.0.2&w=680&fit=max&v=1674050419&gif-q=50&lossless=true&s=fcf9bb993fc194fead7373814aff065d)

Our code isn't the only thing that's open and moddable. Our character assets are specifically designed for easy customization and remixing, complete with [art tutorials](https://fishfolk.github.io/jumpy-legacy-docs/contributing_art.html) on how to match our style conventions.

### Open business practices

Making a commercial game is hard enough. Making a commercial yet open game presents many additional challenges, which is why it's so unheard of. But we believe it's worth the added difficulty, for the sake of building something timeless.

### Community centric

We may be novice game developers (some of us have industry experience, but we're by no means veterans), but community is our crab & butter. We've been building professional game communities for decades, and Fish Folk is essentially our magnum opus.

# World of Fish Folk

![](https://ksr-ugc.imgix.net/assets/039/700/667/bba7d2dffa9ff2d72a1af5efcb79819c_original.png?ixlib=rb-4.0.2&w=680&fit=max&v=1674047426&gif-q=50&lossless=true&s=54694f98ecb752bf84479201400e9a88)

Uhm, did we mention we're making multiple games in one? The [World of Fish Folk](https://spicylobster.itch.io/jumpy/devlog/412030/welcome-to-the-world-of-fish-folk) is an ever-expanding bundle of games!

Inspired by the much celebrated [Kunio-kun](https://en.wikipedia.org/wiki/Kunio-kun) series from our childhood, we don't want to limit our fancy fish characters to just one game title.

The [Fish World Asset Pack](https://spicylobster.itch.io/fish-world-pack) is freely available for anyone to use as CC BY-NC licensed content, so other fish games may easily be prototyped and proposed for official inclusion in our series.

The flagship game we're presenting front end center in this campaign is technically 'Fish Folk: Jumpy', which is now in a fully playable v1 alpha stage. By the end of 2023, we expect to have completed four titles

-   [Jumpy](https://github.com/fishfolks/jumpy) v0.5_alpha (2D platformer, where it all began)
    
-   [Punchy](https://github.com/fishfolks/punchy) v0.2.1_pre-alpha(2D beatemup)
    
-   [Bomby](https://github.com/fishfolk/bomby) v0.1.0_pre-alpha (2D bomber arena)
    
-   [Ballsy](https://github.com/fishfolks/ballsy) 0.1.0-pre_alpha (2D ball-games)
    

Fish Folk isn’t a game, it’s a whole bunch of games. What these games have in common is that they are..

1.  all in the same universe; wacky fish in the style of Nintendofied Spongebob;
    
2.  all really easy to play; no prior gaming experience required;
    
3.  best played in groups of 2-4 people, locally🌟  or online⭐️ .
    

# Paying it forward

This project is ultimately about continuing the four-decade long tradition of making [awesome platformer games](https://en.wikipedia.org/wiki/Platform_game), all standing on the shoulders of giants before them. In the [words of superjoebob](https://superjoebob.wordpress.com/2019/04/30/thank-you/):

"I’ve worked 5 years on Duck Game, but it’s not really ‘mine’. It’s about as much ‘my’ game as Hotline Miami, Nidhogg, Super Mario World and the hundreds of other games that inspired its existence are."

We're honoring this tradition in the best way we know how, by making a foundation so sturdy that it may last for another four decades. 

A continuously evolving *[world turtle](https://en.wikipedia.org/wiki/World_Turtle)*, with shoulders strong enough to support an ecosystem far bigger than itself.

![0FE2B632-EC30-4110-82B5-5A3992320A24](https://user-images.githubusercontent.com/583842/221295484-1a774b42-d26f-4e66-bd04-58d3b43912c4.png)

**Comment by zicklag** [comment]
I like it, well done. I think it gives a great impression of the vision and value. 👍 🚀

**Comment by orhun** [comment]
Awesome read, I can really sense the motivation behind it. Looking forward to the Kickstarter campaign. 🚀🦀

**Comment by erlend-sh** [comment]
Drafting the video script, which incidentally can also be used as a  ‘whole pitch in 500 words’ preamble to the main pitch:

- - -

[Open with idle fish screen.]

Fish Folk is a spiritual successor to the classic Duck Game, developed open source with longevity and community in mind.
[Duck Game couch play]

a 2-4 player 2D arena shooter with lots of emergent gameplay.
[quick Duck Game highlights]

We’ve been working on this game for 3+ years. It was playable since the first ~3 months of development, but we’ve gone through a handful of iterations to arrive at our current state.

The game should last forever, therefore we are making it:
- community-centric; cooperatively owned.
- super moddable (WASM-scripting, procgen tools, creators platform)
- with Rust-lang, Bevy engine and Bones.
- peer-to-peer networked ([ggpo rollback](https://en.wikipedia.org/wiki/GGPO)), i.e. self-hostable
- an [open source project](https://github.com/fishfolk/jumpy)

The demo contains:
- 12+ maps for 2-4 players
- Opponent AI bots
- Grenades, mines, seaweed, sproingers, spike shoes..
- Dynamic characters (essential wearables, emotes & animations)
- Level editor

The released game will feature roughly a 5x increase of the demo’s contents. Also in the pipeline:

- Lots of polish & game-feel tweaking
- GFX, lighting, particles etc.
- Automagic level editing; procedural generation
- Online multiplayer
- Tournaments & leaderboards
- Modding platform

ALSO, Fish Folk isn’t just a game, it’s a universe. We plan to expand the world of Fish Folk with many classic game archetypes, all in one, ever-expanding bundle. Already in the works: A beat-em-up, arcade soccer and bomber-fish.

The big picture of this thing more than just a game. We're trying to build the open source modding platforms of our dreams.

In Warcraft 3, modders didn't even get to own any of their creations, and monetization was never an option. With the likes of Roblox we've finally gotten to the point where community creators can be commercially successful, but their ecosystem still mimics the worst parts of the modern marketplace, with only a few big fish eating most of the pie.

We believe the key to a healthier modding ecosystem is openness, so we can shape the virtual worlds we want to live in together.

**Comment by zicklag** [comment]
## Modding Gifs

I'm working on some modding Gifs. Here's the first:

### Hot Reload Sprites
![hot-reload-sprites](https://user-images.githubusercontent.com/25393315/224063807-311c88c1-bc79-4fe0-94f2-7f0fa048d9cb.gif)

**Reply by zicklag** [reply]
### Hot Reload YAML Config

![yaml-modding3](https://user-images.githubusercontent.com/25393315/224067982-cfbd2fab-d736-471a-a3ec-ed8084e2ccee.gif)

**Reply by zicklag** [reply]
### Live Tile Editing

![live-tile-editing5](https://user-images.githubusercontent.com/25393315/224072352-38644831-4b27-4621-9b82-0e64a86c803c.gif)

**Reply by erlend-sh** [reply]
Note for future: Doing a sprite editing clip with [Pixelorama](https://github.com/Orama-Interactive/Pixelorama) some time would be cool as well. It's got a slightly more modern look, quite similar to our native editor.

**Reply by zicklag** [reply]
Oh, yeah, it'd be easy to do that. I'll probably do that tomorrow when I update the release notes.

**Reply by zicklag** [reply]
Maybe we should make a VSCode theme to match Jumpy sometime, too, just for fun. :)

**Comment by erlend-sh** [comment]
Our previous trailer from years back:

https://user-images.githubusercontent.com/583842/224064201-93fb4dad-8263-4560-a150-61ada3f72cff.mp4
